packages/web/utils/stringHelpers.ts

```diff
@@ -7,3 +11,57 @@ export const hashCode = (str: string): string => {
   }
   return hash.toString();
 };
+
+/**
+ * sanitize string of HTML & parse it into jsx elements
+ * */
+export const safelyParseContent = (content: string) => {
+  const clean = DOMPurify.sanitize(content);
+  const options = {
+    replace: (domNode: any) => {
+      if (domNode.attribs?.href) {
+        domNode.attribs.target = '_blank';
+        domNode.attribs.title = `Opens new tab to ${domNode.attribs.href}`;
+        domNode.attribs.class = 'external-link';
+      }
+    },
+  };
+
+  const parsed = parse(clean, options);
+  return parsed;
+};
+
+/**
+ * Takes a string of HTML elements, sanitizes it,
+ * applies classNames to the elements & returns parsed HTML as
+ * `JSX.Element | JSX.Element[] | string` */
+export const safelyParseTextForTyping = (content: string) => {
+  const clean = DOMPurify.sanitize(content);
+  const options = {
+    replace: (domNode: any) => {
+      if (domNode.type === 'tag' && domNode.children.length > 0) {
+        if (domNode.parent === null) {
+          domNode.attribs.class = 'typing-text';
+        }
+        if (domNode.attribs?.href) {
+          domNode.attribs.target = '_blank';
+          domNode.attribs.title = `Opens new tab to ${domNode.attribs.href}`;
+          domNode.attribs.class = 'external-link';
+        }
+      }
+    },
+  };
+
+  const parsed = parse(clean, options);
+  return parsed;
+};
+
+export const shortenAddress = (address: string, sliceSize?: number): string => {
```

Comment:
there's a formatAddress in `web3/utils/playerHelpers.ts` that does something similar to this and perhaps we could reuse that here, or modify that method to support this extra paramer "sliceSize".

Reply:
> there's a formatAddress in `web3/utils/playerHelpers.ts` that does something similar to this and perhaps we could reuse that here, or modify that method to support this extra paramer "sliceSize".

I missed that. Modifying the method sounds like the most sensible thing to do. thanks.

Reply:
✅